targets/arty/base.py

```diff
@@ -162,7 +160,7 @@ def __init__(self, platform, spiflash="spiflash_1x", **kwargs):
                 endianness=self.cpu.endianness)
         self.add_constant("SPIFLASH_PAGE_SIZE", 256)
         self.add_constant("SPIFLASH_SECTOR_SIZE", 0x10000)
-        self.add_wb_slave(mem_decoder(self.mem_map["spiflash"]), self.spiflash.bus)
```

Comment:
I'm not sure about all the changes removing `mem_decoder`.

The problem with it is that the function returns lambda function instead of a number, so it's not even type compatible with `add_wb_slave`.

What was the reason of using `mem_decoder` in the first place?

Reply:
https://github.com/timvideos/litex-buildenv/pull/311